test/Test/Data/Mutable/Array.hs

```diff
@@ -67,6 +51,9 @@ group =
   , testProperty "∀ a,i,x. len (write a i x) = len a" lenWrite
   , testProperty "∀ a,s,x. len (resize s x a) = s" lenResizeSeed
   -- Tests against a reference implementation
+  , testProperty "∀ a,ix. write a ix . fromList = fromList . write a ix" writeRef
+  , testProperty "∀ ix. read ix (fromList l) = l !! i" readRef
+  , testProperty "size . fromList = length" sizeRef
```

Comment:
These properties are better specified in terms to `toList` than in term of `fromList`.

Despite the fact that we actually use `fromList` for the test, because we don't have a choice. Using `fromList` in the specification implies that arrays which are not of the form `fromList l` are exempt from the rule. There are, of course, non, or at least so we claim. But it should make it clear that `toList` is the more economical direction.

Another way to think about it (the right way if you are me), is that this is a property of arrays, so should quantify over the array, it is written in terms of the semantics of arrays, which is given by `toList`.

Just to make sure that there is no misunderstanding: I'm only asking you to rewrite the test names in the strings. The test themselves are fine.

Reply:
I just assumed that `fromList` was onto it's domain (which it is, though for vectors, it isn't). I'll re-write the test strings :+1: